src/fileparse.py

```diff
@@ -66,9 +78,10 @@ def parse_csv(
             rowdict = dict(zip(headers, row))
             if convert_fn:
```

Comment:
I think you can replace this check with the `isinstance(convert_fn, dict)` and remove this below